utils/hwloc/misc.h

```diff
@@ -946,20 +947,69 @@ hwloc_utils_parse_cpuset_format(const char *string)
     return HWLOC_UTILS_CPUSET_FORMAT_HWLOC;
   else if (!strcmp(string, "list"))
     return HWLOC_UTILS_CPUSET_FORMAT_LIST;
+  else if (!strcmp(string, "systemd-dbus-api"))
+    return HWLOC_UTILS_CPUSET_FORMAT_SYSTEMD;
   else if (!strcmp(string, "taskset"))
     return HWLOC_UTILS_CPUSET_FORMAT_TASKSET;
   else
     return HWLOC_UTILS_CPUSET_FORMAT_UNKNOWN;
 }
 
+/* The AllowedCPUs systemd DBus API syntax expects a string as follows:
+ * "ay 0xNNNN 0xAA [0xBB [...]]"
+ * where:
+ *   "0xNNNN" is the size of the array, max size 2^26.
+ *   "0xAA [0xBB [...]]" is the cpuset mask given bytes after bytes, little endian order.
+ * e.g. the output of `hwloc-calc --cpuset-input-format list 0,31-32,63-64,77 --cpuset-output-format systemd-dbus-api` is
+ *   "AllowedCPUs ay 0x0a 0x01 0x00 0x00 x80 0x01 0x00 0x00 0x80 0x01 0x20
+ * */
+#define HWLOC_SYSTEMD_ALLOWEDCPUS_PREFIX_FORMAT "AllowedCPUs ay 0x%02x"
+#define HWLOC_SYSTEMD_ALLOWEDCPUS_PREFIX_SIZE 19
+#define HWLOC_SYSTEMD_ALLOWEDCPUS_BYTES_FORMAT " 0x%02x"
+#define HWLOC_SYSTEMD_ALLOWEDCPUS_BYTES_SIZE 5
+static __hwloc_inline int hwloc_utils_systemd_asprintf(char ** strp, const struct hwloc_bitmap_s * __hwloc_restrict set)
+{
+  int last = hwloc_bitmap_last(set);
+  if (last == -1 )
```

Comment:
abort() is too strong in this case. what should we do if the cpuset is empty? return an error and document that empty cpuset aren't supported?

Reply:
For the systemd-dbus-apo case, it makes no sense not to pass a cpuset, so we could report an error: `Empty and infinite CPU sets are not supported with the systemd-dbus-api cpuset output format`. 

Can you confirm that `-1` can be any of both: empty or infinite?

Reply:
Yes, both empty and infinite.

Reply:
Fixed